docs/guide/introduction.md

```diff
@@ -16,7 +16,84 @@ In short, it recognizes pain points on both sides of the CMS experience and reli
 
 ## The module system
 
-Vestibulum id ligula porta felis euismod semper. Maecenas faucibus mollis interdum. Cras mattis consectetur purus sit amet fermentum. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus. Donec ullamcorper nulla non metus auctor fringilla. Cras mattis consectetur purus sit amet fermentum. Donec ullamcorper nulla non metus auctor fringilla.
+As you learn to use ApostropheCMS, you'll quickly see how all server-side code is organized in a system of "modules." Modules are the building blocks we use to build Apostrophe projects — as well as the CMS itself.
+
+Each module defines a specific set of functionality, from governing the database connection to setting the fields in a blog post. Common Apostrophe project modules will define custom content types, page types, or editable widget types.
+
+All modules use [the same API](/reference/module-api/). That shared foundation means that you can give your blog post module powerful features, such as custom command line tasks and API routes. It's the same foundation the core team uses to build the CMS, so it is well tested and designed to be as intuitive as possible.
+
+### Setting up a module
+
+Modules are organized in a folder at the root of a project appropriately named `modules`. Each module gets its own directory inside that with an `index.js` file that contains its configurations. So the blog post code would being in the file:
+
+```
+modules/blog-post/index.js
+```
+
+The module code itself would then be defined in an object, assigned to `module.exports`. This may look familiar if you know [CommonJS](https://nodejs.org/api/modules.html#modules_modules_commonjs_modules) patterns.
+
+```javascript
+// modules/blog-post/index.js
+module.exports = {
+  // ...
+}
+```
+
+The final step to this minimalist module example is to tell Apostrophe that it should be turned on, or instantiated. That is done in the main application file, `app.js` in its own `modules` property.
+
+```js
+// app.js
+require('apostrophe')({
+  modules: {
+    'blog-post': {}
+  }
+});
+```
+
+There is more to do to make modules useful, but that is all it takes to get them running.
+
+### Module inheritance
+
+Inheritance is the glue of the module system. Every module, other than the root module, extends another one. This means that your blog post module, which extends the "piece type" module, has a huge set of features you never have to write.
```

Comment:
Link to piece type module when we have a chance. Or this might be a future opportunity for a tooltip to go over "piece type"